src/utils/httpRequest.ts

```diff
@@ -23,18 +33,37 @@ export async function httpRequest<T>(url: string, options?: RequestOptionsLike,
 
     const response = await fetch(request);
 
-    if (response.status >= 200 && response.status < 300) {
-        return new HttpResponse(response);
+    if (errorHandling === ErrorHandling.ReturnErrorResponse || response.ok) {
+        return new HttpResponse(response, url);
     } else {
         throw new HttpErrorResponse(response);
     }
 }
 
-export class HttpResponse<T> {
+export class HttpResponse<T> implements ResponseLike {
     private bodyPromise: Promise<T> | undefined;
-    private normalizedHeaders: { [key: string]: string } | undefined;
+    public readonly headers: HeadersLike;
+    public readonly status: number;
+    public readonly statusText: string;
+    public readonly ok: boolean;
+
+    public constructor(private readonly innerResponse: Response, public readonly url: string) {
+        // Unfortunately Typescript will not consider a getter accessor when checking whether a class implements an interface.
```

Comment:
That's surprising!

Reply:
Yes, that's unfortunate. Also adding a setter accessor, or prepending interface members with `readonly` does not help either. 😞